core/src/main/java/com/amplifyframework/api/graphql/GraphQLResponse.java

```diff
@@ -139,12 +199,19 @@ public boolean equals(Object thatObject) {
 
             Error error = (Error) thatObject;
 
-            return message.equals(error.message);
+            return ObjectsCompat.equals(message, error.message) &&
+                   ObjectsCompat.equals(path, error.path) &&
+                   ObjectsCompat.equals(extensions, error.extensions) &&
+                   ObjectsCompat.equals(locations, error.locations);
         }
 
```

Comment:
Need to update `GraphQLResponse.Error`'s `toString()` as well, I think. Right now it just only prints the `message` , but not any of your new fields.

Reply:
Will do!